src/mechanics/constraintsPde/ConstraintCompanion.cpp

```diff
@@ -0,0 +1,125 @@
+#include "mechanics/constraintsPde/ConstraintCompanion.h"
+#include "mechanics/nodes/NodeSimple.h"
+#include "mechanics/constraintsPde/Equation.h"
+
+namespace NuTo
+{
+namespace ConstraintPde
+{
+
+std::vector<Equation> Component(const NodeSimple& node, std::vector<eDirection> directions, double value)
+{
+    return Component(node, directions, RhsConstant(value));
+}
+
+
+std::vector<Equation> Component(const NodeSimple& node, std::vector<eDirection> directions, RhsFunction rhs)
+{
+    std::vector<Equation> eqs;
+    for (auto direction : directions)
+    {
+        int component = ToComponentIndex(direction);
+        eqs.emplace_back(node, component, rhs);
+    }
+    return eqs;
+}
+
+
+std::vector<Equation> Component(const Groups::Group<NodeSimple>& nodes, std::vector<eDirection> directions,
+                                double value)
+{
+    return Component(nodes, directions, RhsConstant(value));
+}
+
+
+std::vector<Equation> Component(const Groups::Group<NodeSimple>& nodes, std::vector<eDirection> directions,
+                                RhsFunction rhs)
+{
+    std::vector<Equation> eqs;
+    for (const auto& node : nodes)
+    {
+        auto tmpEqs = Component(node, directions, rhs);
+        eqs.insert(eqs.end(), tmpEqs.begin(), tmpEqs.end());
+    }
+    return eqs;
+}
+
+
+Equation Direction(const NodeSimple& node, Eigen::VectorXd direction, RhsFunction rhs)
+{
+
+    assert(direction != Eigen::VectorXd::Zero(direction.rows()) && "the direction vector contains only zeros");
+
+    // Normalization necessary, otherwise rhs depends on length of the direction vector
+    direction.normalize();
+
+    int firstNonZeroComp = -1;
```

Comment:
I would prefer using `direction.cwiseAbs().maxCoeff(&maxComponent)` instead of this loop/if thing. First, it is shorter. Second, it is more reasonable. In the gauss elimination we had before, we also chose the highest coefficient in the equation. 
In the loop below you additionally check (`iComponent != maxComponent`).

Reply:
As we discussed before, this is an option too and I ll just do that.